File: task_1/task_1.py
def sum_char(s):
    """Считает количество букв в строке."""
    result = {}
    for char in s:
        
        result[char] = result.get(char, 0) + 1 # get(char, 0) вернет 0, если ключа еще нет в словаре
    return result

def check_palindrome(char_map):
    """Проверяет, можно ли из этих букв составить палиндром."""
    odd_count = 0
    for count in char_map.values():
        if count % 2 == 1:
            odd_count += 1
            if odd_count > 1:
                return False
    return True

def build_palindrome(char_map):
    """Строит сам палиндром."""
    left = []
    middle = ""
    
    
    for char in sorted(char_map.keys()): # Сортируем ключи
        count = char_map[char]
        left.append(char * (count // 2)) # Добавляем половину символов в левую часть
        if count % 2 == 1: # Если количество нечетное, этот символ пойдет в середину
            middle = char
            
    left_str = "".join(left)
    return left_str + middle + left_str[::-1] # В Python разворот строки делается через срез [::-1]

def main():
    old_str = input().strip()
    char_map = sum_char(old_str)
    
    if not check_palindrome(char_map):
        print("No")
    else:
        print("Yes")
        print(build_palindrome(char_map))

if __name__ == "__main__":
    main()

    # python3 task_1.py

File: task_1/task_1.kt
import java.util.*

// Функция для подсчета количества букв
fun sumChar(str: String): Map<Char, Int> {
    val result = mutableMapOf<Char, Int>()
    for (char in str) {
        // getOrDefault позволяет избежать проверки на null
        result[char] = result.getOrDefault(char, 0) + 1
    }
    return result
}

// Проверка, можно ли составить палиндром
fun check(counts: Map<Char, Int>): Boolean {
    var oddCount = 0
    for (count in counts.values) {
        if (count % 2 == 1) {
            oddCount++
            if (oddCount > 1) {
                return false
            }
        }
    }
    return true
}

// Построение палиндрома
fun buildPalindrome(counts: Map<Char, Int>): String {
    val left = StringBuilder()
    var middle = ""

    // Проходим по отсортированной карте (TreeMap), чтобы результат был стабильным
    val sortedCounts = counts.toSortedMap()

    for ((char, count) in sortedCounts) {
        // Добавляем половину символов в левую часть
        repeat(count / 2) {
            left.append(char)
        }
        // Если количество нечетное — этот символ пойдет в середину
        if (count % 2 == 1) {
            middle = char.toString()
        }
    }

    val leftPart = left.toString()
    val rightPart = leftPart.reversed()

    return leftPart + middle + rightPart
}

fun main() {
    val scanner = Scanner(System.`in`)
    
    if (!scanner.hasNext()) return
    val oldStr = scanner.next()
    
    val charCounts = sumChar(oldStr)

    if (!check(charCounts)) {
        println("No")
    } else {
        println("Yes")
        println(buildPalindrome(charCounts))
    }
}

// чтобы скомпилировать kotlinc hello.kt -include-runtime -d hello.jar и запустить java -jar hello.jar